packages/material-ui/src/Theme/MaterialUIContext.tsx

```diff
@@ -1,92 +1,130 @@
-import React from 'react';
-import MaterialUIContextProps from './MaterialUIContextProps';
+import React, { ComponentType } from 'react';
+import { BoxProps, ButtonProps } from '@material-ui/core';
+import MaterialUIContextProps, { defaultMaterialUIContext } from './MaterialUIContextProps';
+
+export const MaterialUIContext: MaterialUIContextProps = defaultMaterialUIContext;
 
-/** Use require for loading these libraries in case they are not available in order to perform a useful fallback
- */
-let mui;
-let icons;
 try {
-  mui = require('@material-ui/core');
-  icons = require('@material-ui/icons');
+  /**
+   * Dynamically import MUI components so we get full tree-shaking support
```

Comment:
Can you transfer this comment's explanation to the comment here? https://github.com/rjsf-team/react-jsonschema-form/issues/2748#issuecomment-1062236301

So it's really clear why we can't use option 2